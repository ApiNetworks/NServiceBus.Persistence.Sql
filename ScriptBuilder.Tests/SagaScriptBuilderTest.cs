using System.Collections.Generic;
using System.IO;
using System.Text;
using ApprovalTests;
using NServiceBus.SqlPersistence;
using NUnit.Framework;

[TestFixture]
public class SagaScriptBuilderTest
{
    [Test]
    public void BuildCreateScript()
    {
        var sagas = new List<SagaDefinition>
        {
            new SagaDefinition
            {
                Name = "theSaga",
                MappedProperties = new List<string>
                {
                    "Property1",
                    "Property2"
                }
            }
        };

        var builder = new StringBuilder();
        using (var writer = new StringWriter(builder))
        {
            SagaScriptBuilder.BuildCreateScript("theschema", "theendpointname", sagas, s => writer);
        }
        var script = builder.ToString();

        SqlValidator.Validate(script);
        Approvals.Verify(script);
    }

    [Test]
    public void BuildDropScript()
    {
        var sagas = new List<string>
        {
            "theSaga"
        };

        var builder = new StringBuilder();
        using (var writer = new StringWriter(builder))
        {
            SagaScriptBuilder.BuildDropScript("theschema", "theendpointname", sagas, s => writer);
        }
        var script = builder.ToString();
        SqlValidator.Validate(script);
        Approvals.Verify(script);
    }
}
