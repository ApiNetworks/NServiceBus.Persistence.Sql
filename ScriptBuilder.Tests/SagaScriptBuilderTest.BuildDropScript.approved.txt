
declare @sagaName nvarchar(max) = 'theSaga';
declare @tableName nvarchar(max) = '[' + @schema + '].[' + @endpointName + '.' + @sagaName + ']';

IF EXISTS 
(
    SELECT * 
    FROM sys.objects 
    WHERE 
        object_id = OBJECT_ID(@tableName) 
        AND type in (N'U')
)
BEGIN
DECLARE @createTable nvarchar(max);
SET @createTable = N'
    DROP TABLE ' + @tableName + '
';
exec(@createTable);
END
